gs_chunked_io/reader.py

```diff
@@ -28,7 +28,15 @@ def __init__(self, blob: Blob, chunk_size: int=default_chunk_size):
     def fetch_chunk(self, chunk_number: int):
         start_chunk = chunk_number * self.chunk_size
         end_chunk = start_chunk + self.chunk_size - 1
-        return self.blob.download_as_string(start=start_chunk, end=end_chunk)
+        if chunk_number == (self.number_of_chunks - 1):
+            expected_part_size = self.blob.size % self.chunk_size
+        else:
+            expected_part_size = self.chunk_size
+        for _ in range(reader_retries):
+            data = self.blob.download_as_string(start=start_chunk, end=end_chunk)
+            if expected_part_size == len(data):
+                return data
+        raise ValueError("Unexpected part size")
```

Comment:
Ooof... I take it this is undocumented?

Reply:
I'm sure they don't _want_ to return partial downloads without error.

This is a very annoying bug.